### ISSUE TITLE
[BUG] Custom hypothesis strategies don't work with explicit examples

### ISSUE THREAD

**Issue by Stranger6667** [schemathesis/schemathesis#1217]
**Describe the bug**
When the schema contains explicit examples (via the `example` keyword), and the test function is wrapped with `@schema.given`, it leads to an internal error.

**To Reproduce**
Start the built-in server with `./test_server.sh 8081 --operations=get_payload`

Run `pytest` against this file
```python
import schemathesis
from hypothesis import strategies as st


schema = schemathesis.from_uri(
    "http://127.0.0.1:8081/schema.yaml",
    validate_schema=False,
)


@schema.parametrize()
@schema.given(data=st.data())
def test_api(data, case):
    response = case.call()
    case.validate_response(response)
```

Observe the error:

```
TypeError: test_api() missing 1 required positional argument: 'data'
```

Explicit examples can be combined with `schema.given`

**Environment:**
 - Schemathesis version: 3.9.3
 - Spec version: Open API 2 / 3

**Additional context**
Reported in [gitter](https://gitter.im/schemathesis/schemathesis?at=60dc8e44b60c3137c39e9305)

**Comment by Stranger6667** [comment]
With the `example` based implementation, it can't work - `example` is constructed by Schemathesis and expects exact values. But if `test_api` uses `data` to draw values interactively, then the user might expect it to work this way even with explicit examples. Though, it is unclear what value should be passed to `example` in such a case.

I think, that if explicit examples would be implemented as a separate test type (#1065), then the issue will be solved. I.e. there will be no `example` and there will be a different flow, that allows drawing interactively in tests

**Comment by Stranger6667** [comment]
Hypothesis rejects such tests at all, probably we need to replicate this behavior here.
